### PR TITLE
Feature/azure/cr 2265 not car azure ensure authenticat

### PR DESCRIPTION
**Pull request by alonashkenazi**

### CODE REVIEW

File: tests/knowledge/rules/azure/non_context_aware/test_app_service_authentication_enable.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    def setUp(self):
+        self.rule = AppServiceAuthenticationEnableRule()
+
+    @parameterized.expand(
+        [
+            ["auth enabled should not alert", True, False],
+            ["auth disabled should alert", False, True]
+        ]
+    )
+    def test_auth_states(self, unused_name: str, auth_enable: bool, should_alert: bool):
```
Comment (Tomerva) [inline tests/knowledge/rules/azure/non_context_aware/test_app_service_authentication_enable.py:26]: remove unused_name

File: cloudrail/knowledge/rules/azure/non_context_aware/app_service_authentication_enable_rule.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+class AppServiceAuthenticationEnableRule(AzureBaseRule):
+
+    def get_id(self) -> str:
+        return 'non_car_authentication_enabled_in_web_app'
+
+    def execute(self, env_context: AzureEnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
+        issues: List[Issue] = []
+        for app_service in env_context.app_services:
+            if app_service.app_service_config is None or \
```
Comment (Tomerva) [inline cloudrail/knowledge/rules/azure/non_context_aware/app_service_authentication_enable_rule.py:17]: how can the config be none? i think it should atleast be initialized with default values

File: cloudrail/knowledge/rules/azure/non_context_aware/app_service_ftps_required_rule.py [hunk 3]
```diff
@@ -14,7 +14,8 @@ def get_id(self) -> str:
     def execute(self, env_context: AzureEnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
         issues: List[Issue] = []
         for app_service in env_context.app_services:
-            if app_service.ftps_state == FtpsState.ALL_ALLOWED:
+            if app_service.app_service_config is None or \
```
Comment (Tomerva) [inline cloudrail/knowledge/rules/azure/non_context_aware/app_service_ftps_required_rule.py:17]: same as above

**Review by Tomerva — APPROVED** [review]

**Review by Viperain — APPROVED** [review]

File: cloudrail/knowledge/context/azure/webapp/azure_app_service_config.py [hunk 4]
```diff
@@ -29,3 +33,7 @@ def get_cloud_resource_url(self) -> Optional[str]:
     @property
     def is_tagable(self) -> bool:
         return False
+
+    @staticmethod
+    def is_standalone() -> bool:
```
Comment (yi2020) [inline cloudrail/knowledge/context/azure/webapp/azure_app_service_config.py:38]: What's this?

Comment (alonashkenazi) [reply cloudrail/knowledge/context/azure/webapp/azure_app_service_config.py:38]: A flag that indicates that this object doesn't stand by itself. we are using it for the drift detection

**Pull request merged**